ReSharper.FSharp/src/FSharp.Psi.Features/src/Daemon/Analyzers/IfExpressionAnalyzer.fs

```diff
@@ -0,0 +1,36 @@
+namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.Analyzers
+
+open JetBrains.ReSharper.Feature.Services.Daemon
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.Highlightings.Errors
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Impl
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Parsing
+open JetBrains.ReSharper.Plugins.FSharp.Psi.PsiUtil
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Tree
+open JetBrains.ReSharper.Plugins.FSharp.Util
+
+[<ElementProblemAnalyzer([| typeof<IIfThenElseExpr> |],
+                         HighlightingTypes = [| typeof<IfCanBeReplacedByConditionOperandWarning> |])>]
+type IfExpressionAnalyzer() =
+    inherit ElementProblemAnalyzer<IIfThenElseExpr>()
+
+    override this.Run(expr, _, consumer) =
+        if not (skipMatchingNodesAfter isWhitespace expr.ThenKeyword == expr.ThenExpr) ||
+           not (skipMatchingNodesAfter isWhitespace expr.ThenExpr == expr.ElseKeyword) ||
+           not (skipMatchingNodesAfter isWhitespace expr.ElseKeyword == expr.ElseExpr) then () else
+
```

Comment:
Do we need to traverse all the expressions surroundings before even checking if we're interested in this `if` expression?

Reply:
This code will fail with NREs as soon as any of the accessed children is missing, breaking the rest of the analysis too. There should be a test with missing `else` branch, and every child accessor should have a `null` check.